libcaf/libcaf/repository.py

```diff
@@ -691,6 +701,231 @@ def create_tag(self, tag_name: str, commit_hash: str, author: str, message: str)
     @requires_repo
     def status(self) -> Sequence[Diff]:
         return self.diff(self.head_commit(), self.working_dir)
+    
+    def _resolve_checkout_target(self, target: str) -> Ref | None:
```

Comment:
I don't see why this function is needed, you already have resolve_ref which determines what kind of ref you're dealing with. Making another function just risks bugs, inconsistencies and increases complexity. Just use resolve_ref.